packages/block-library/src/calendar/edit.js

```diff
@@ -61,7 +61,16 @@ class CalendarEdit extends Component {
 }
 
 export default withSelect( ( select ) => {
+	const {
+		getEditedPostAttribute,
+	} = select( 'core/editor' );
+	const postType = getEditedPostAttribute( 'type' );
+	// Dates are used to overwrite year and month used on the calendar.
+	// This overwrite should only happen for 'post' post types.
+	// For other post types the calendar always displays the current month.
 	return {
-		date: select( 'core/editor' ).getEditedPostAttribute( 'date' ),
+		date: postType === 'post' ?
+			getEditedPostAttribute( 'date' ) :
```

Comment:
Question: Why is the block dependent on the current post being edited? Is the widget dependent on posts too? Doesn't seem like it since the widget can be used in pages without any post?

Reply:
It's complex, see my investigation summarized in this comment: https://github.com/WordPress/gutenberg/pull/13772#discussion_r256775347.

Reply:
Hi @youknowriad, In fact, the widget depends on the post and shows the calendar of the post publish date the widget is being shown in.

Reply:
This is a summary of the research I did while implementing the block:
The calendar widget renders the calendar of the month the current post was published, if the widget is an archives page it renders the calendar of that month, and if it is a neutral non-date specific page (homepage, taxonomy, author, and as @gziolo referred page and other cpts) it renders the calendar of the current month.

Reply:
Got it :) I guess we'll have to revisit this once we start the widgets page as there won't be any `getEditedPostAttribute` selector there.